# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/Layers.h
# Commit: bbfe23c61add
# Full Hash: bbfe23c61add07377daf80696fd6996c0463f5b6
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-09 09:21:34
# Regressor Bug: 1632249
# File Overlap Count: 3
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r 8fcfac450ecc -r bbfe23c61add gfx/layers/Layers.h
--- a/gfx/layers/Layers.h	Mon Jun 08 20:13:01 2020 +0000
+++ b/gfx/layers/Layers.h	Mon Jun 08 20:34:15 2020 +0000
@@ -77,7 +77,6 @@
 namespace layers {
 
 class Animation;
-class AsyncCanvasRenderer;
 class AsyncPanZoomController;
 class BasicLayerManager;
 class ClientLayerManager;
@@ -2540,7 +2539,7 @@
 
   const nsIntRect& GetBounds() const { return mBounds; }
 
-  CanvasRenderer* CreateOrGetCanvasRenderer();
+  RefPtr<CanvasRenderer> CreateOrGetCanvasRenderer();
 
  public:
   /**
@@ -2581,9 +2580,9 @@
   void DumpPacket(layerscope::LayersPacket* aPacket,
                   const void* aParent) override;
 
-  virtual CanvasRenderer* CreateCanvasRendererInternal() = 0;
+  virtual RefPtr<CanvasRenderer> CreateCanvasRendererInternal() = 0;
 
-  UniquePtr<CanvasRenderer> mCanvasRenderer;
+  RefPtr<CanvasRenderer> mCanvasRenderer;
   gfx::SamplingFilter mSamplingFilter;
 
   /**